src/plugins/console/server/routes/api/console/autocomplete_entities/index.ts

```diff
@@ -6,4 +6,194 @@
  * Side Public License, v 1.
  */
 
-export { registerMappingsRoute } from './register_mappings_route';
+import http from 'http';
+import https from 'https';
+import { Buffer } from 'buffer';
+import { parse } from 'query-string';
+import Boom from '@hapi/boom';
+import type { KibanaRequest } from '@kbn/core-http-server';
+import type { SemVer } from 'semver';
+import type { RouteDependencies } from '../../..';
+import { sanitizeHostname } from '../../../../lib/utils';
+import type { ESConfigForProxy } from '../../../../types';
+import { getRequestConfig } from '../proxy/create_handler';
+
+interface Settings {
+  indices: boolean;
+  fields: boolean;
+  templates: boolean;
+  dataStreams: boolean;
+}
+
+type Config = ESConfigForProxy & { headers: KibanaRequest['headers'] } & { kibanaVersion: SemVer };
+
+const MAX_RESPONSE_SIZE = 10 * 1024 * 1024; // 10MB
+// Limit the response size to 10MB, because the response can be very large and sending it to the client
+// can cause the browser to hang.
+
+const getMappings = async (settings: Settings, config: Config) => {
+  if (settings.fields) {
+    const mappings = await getEntity('/_mapping', config);
+    return mappings;
+  }
+  // If the user doesn't want autocomplete suggestions, then clear any that exist.
+  return {};
+};
+
+const getAliases = async (settings: Settings, config: Config) => {
+  if (settings.indices) {
+    const aliases = await getEntity('/_alias', config);
+    return aliases;
+  }
+  // If the user doesn't want autocomplete suggestions, then clear any that exist.
+  return {};
+};
+
+const getDataStreams = async (settings: Settings, config: Config) => {
+  if (settings.dataStreams) {
+    const dataStreams = await getEntity('/_data_stream', config);
+    return dataStreams;
+  }
+  // If the user doesn't want autocomplete suggestions, then clear any that exist.
+  return {};
+};
+
+const getLegacyTemplates = async (settings: Settings, config: Config) => {
+  if (settings.templates) {
+    const legacyTemplates = await getEntity('/_template', config);
+    return legacyTemplates;
+  }
+  // If the user doesn't want autocomplete suggestions, then clear any that exist.
+  return {};
+};
+
+const getIndexTemplates = async (settings: Settings, config: Config) => {
+  if (settings.templates) {
+    const indexTemplates = await getEntity('/_index_template', config);
+    return indexTemplates;
+  }
+  // If the user doesn't want autocomplete suggestions, then clear any that exist.
+  return {};
+};
+
+const getComponentTemplates = async (settings: Settings, config: Config) => {
+  if (settings.templates) {
+    const componentTemplates = await getEntity('/_component_template', config);
+    return componentTemplates;
+  }
+  // If the user doesn't want autocomplete suggestions, then clear any that exist.
+  return {};
+};
+
+const getEntity = (path: string, config: Config) => {
```

Comment:
Can we add a code comment here to explain why we are using the raw http request to retrieve the entities instead of esClient?